### PR TITLE
Add snapshot tests and docs

### PR DESCRIPTION
**Pull request by SimonCropp**
So this does several things. Please let me know if u want any of it split out into individual PRs. Or just close if u dont like the approaches :)

TLDR: here is the [resulting readme](https://github.com/SimonCropp/BlazorSliders/blob/stuff/README.md), the [resulting tests](https://github.com/SimonCropp/BlazorSliders/blob/stuff/UnitTests/IntegrationTest.cs), and an [example screenshot from the test](https://github.com/SimonCropp/BlazorSliders/blob/stuff/UnitTests/IntegrationTest.Crazy.01.verified.png)

## Snapshot testing

 * Add snapshot tests of the page using [Verify](https://github.com/VerifyTests/Verify) and [Verify.Playwright](https://github.com/VerifyTests/Verify.HeadlessBrowsers#playwright-usage). this results in a screen capture and the html for any page verified
 * Uses [Verify.AngleSharp](https://github.com/VerifyTests/Verify.AngleSharp) to pretty print the resulting html. this is helpful if u minify html, or compose it in a way that is difficult to keep the indentation correct
 * Uses [Verify.ImageMagick](https://github.com/VerifyTests/Verify.ImageMagick) to provide fuzzy matching. on different machine screencapture can result in slightly different images. VerifyImageMagick enabled lossy comparison of images to that minor pixel issues dont cause a failure of a test
 * There is some [scrubbing](https://github.com/VerifyTests/Verify/blob/master/docs/scrubbers.md) of the resulting html to remove some noise. also removes 

## docs

 * uses [MarkdownSnippets](https://github.com/SimonCropp/MarkdownSnippets) to merge the razor files and snippets for AddSliderInterop. now on every build, those snippets in the readme will be updated with what is currently in the actual code
 * adds a [MarkdownSnippets github action](https://github.com/SimonCropp/MarkdownSnippets/blob/master/docs/github-action.md) that will update the readme if u edit source online, or forget to doa  build before pushing.
 * The readme now has a table of contents, that will be kept up to date based on the headings in the markdown
 * I added images (from the above snapshots of screens) to the readme.
 * Added an icon from the noun project
 * Added a nuget badge

### CODE REVIEW

**Comment by SimonCropp** [comment]
oh and this should be squashed on merge

**Comment by carlfranklin** [comment]
Good stuff! I'm new to Verify and snapshot testing but it seems like a very reasonable approach for web testing.
